service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -269,6 +269,25 @@ public List<Workspace> getWorkspacesMatchingList(List<UUID> idList, int offset,
     return jdbcTemplate.query(sql, params, WORKSPACE_ROW_MAPPER);
   }
 
+  @ReadTransaction
+  public UUID getUuidByUserFacingId(String userFacingId) {
+    if (userFacingId == null || userFacingId.isEmpty()) {
+      throw new MissingRequiredFieldException("userFacingId is required");
+    }
+    String sql = "SELECT workspace_id FROM workspace WHERE user_facing_id = :user_facing_id";
+    MapSqlParameterSource params = new MapSqlParameterSource().addValue("user_facing_id", userFacingId);
+    try {
+      UUID result =
+              DataAccessUtils.requiredSingleResult(
+                      jdbcTemplate.query(sql, params, UUID_ROW_MAPPER));
+      logger.info("Retrieved workspace uuid {}", result);
+      return result;
+    } catch (EmptyResultDataAccessException e) {
+      throw new WorkspaceNotFoundException(
+              String.format("Workspace %s not found.", userFacingId));
```

Comment:
nit: add `userFacingId` to the log messages, since some UFIDs can look like GUIDs.